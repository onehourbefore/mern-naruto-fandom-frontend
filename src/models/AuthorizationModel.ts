import { StatusModel } from "./StatusModel"

export type UserDataType = {
    id: string,
    email: string,
    name: string,
    role: string,
    avatar: string,
    created: number,
    liked: string [] | []
}

export type ResponseType = {
    accessToken: string,
    refreshToken: string,
    user: UserDataType
}

export interface AuthorizationState {
    user: UserDataType,
    status: StatusModel,
    deleteProfileStatus: StatusModel
}