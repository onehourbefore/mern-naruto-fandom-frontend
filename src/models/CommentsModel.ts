import { StatusModel } from "./StatusModel"

export type CommentType = {
    _id: string,
    postID: string,
    date: string,
    name: string,
    email: string,
    body: string,
    avatar: string,
}

export interface CommentsState {
    comments: [] | CommentType [],
    status: StatusModel,
}

export type CommentProps = {
    data: {
        _id: string,
        name: string,
        email: string,
        body: string,
        avatar: string,
        date: string,
    },
    postID: string | undefined,
    commentIndex: number,
}

export type CommentFormProps = {
    setVisible: (arg: boolean) => void,
}

export type CommentFromSidebarProps = {
    comment: CommentType,
}