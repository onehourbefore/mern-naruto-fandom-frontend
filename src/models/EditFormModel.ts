

export enum TasksEnum {
    CREATE= 'create',
    UPDATE = 'update',
    IDLE = 'idle'
}

export interface EditFormState {
    postID: string
    task: TasksEnum
}