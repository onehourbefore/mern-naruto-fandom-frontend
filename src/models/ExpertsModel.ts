import { StatusModel } from "./StatusModel"

export type ExpertType = {
    avatar: string,
    email: string,
    name: string,
    created: number
}


export interface ExpertsState {
    experts: [] | ExpertType [],
    status: StatusModel
}