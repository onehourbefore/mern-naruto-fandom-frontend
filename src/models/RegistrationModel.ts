import { StatusModel } from "./StatusModel"

export type RegistrationFormType = {
    email: string,
    password: string,
    name: string,
    secret: string,
    avatar: string | Blob
}

export interface RegistrationState {
    status: StatusModel
}