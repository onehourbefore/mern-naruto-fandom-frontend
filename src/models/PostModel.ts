import { StatusModel } from "./StatusModel"
import { CommentType } from "./CommentsModel"

export type PostType = {
    _id: string,
    author: string,
    title: string,
    content: string,
    tags: string [],
    likes: number,
    comments: [] | CommentType [],
    views: number,
    image: string
}

export interface PostsState {
    all: { posts: PostType [] | [], status: StatusModel },
    created: { post: PostType | null, status: StatusModel },
    updated: { post: PostType | null, status: StatusModel },
    deleted: { post: PostType | null, status: StatusModel },
    count: { count: number, status: StatusModel }
}

export interface OnePostState {
    active: { post: PostType | null, status: StatusModel },
    likeStatus: StatusModel,
    commentStatus: StatusModel
}

export type FeedbackPanelProps = {
    activePost: PostType,
    id: string | undefined
}

export type PostProps = {
    post: PostType
}

export type PostInfoProps = {
    data: {
        likes: number,
        comments: [] | CommentType [],
        views: number
    }
}

export type PostContentProps = {
    data: {
        author: string,
        title: string,
        content: string,
        tags: string []
    }
}

export type PostDataProps = {
    activePost: PostType
}

export type PostTypeForCreate = {
    title: string,
    content: string,
    tags: string,
    image: Blob | string,
}


