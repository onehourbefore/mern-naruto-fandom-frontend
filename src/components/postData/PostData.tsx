import React from 'react'
import { PostDataProps } from '../../models/PostModel'
import cl from './PostData.module.scss'


const PostData: React.FC <PostDataProps> = (
    { activePost }
): JSX.Element => {
    const { tags, title, content, author } = activePost;
    const tagsItems = tags.map(item => 
        <span 
            className={cl.root__post_author_tags}
            key={item}
        >
            {item}
        </span>);

    return (
        <div className={cl.root__post}>
            <div className={cl.root__header}>
                <div className={cl.root__post_author}>
                    {tagsItems}
                </div>
                <div className={cl.root__post_author}>
                    Автор: {author}
                </div>
            </div>
            <h2 className={cl.root__post_title}>
                {title}
            </h2>
            <div className={cl.root__post_content}>
                {content}
            </div>
        </div>
    )
};

export default PostData;