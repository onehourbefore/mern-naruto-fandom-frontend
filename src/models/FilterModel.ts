import { StatusModel } from "./StatusModel"
import { PostType } from "./PostModel"

export interface FilterState {
    list: [] | PostType [],
    tags: [] | string [],
    fetchStatus: StatusModel,
    searchStatus: StatusModel
};