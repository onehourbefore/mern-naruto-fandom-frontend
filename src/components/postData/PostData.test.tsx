import { render, screen } from '@testing-library/react';
import { PostType } from '../../models/PostModel';
import PostData from './PostData';



const activePost: PostType = {
    tags: ['one', 'two'],
    title: 'ANIME',
    content: `Lorem ipsum dolor sit, amet consectetur adipisicing elit. In quisquam 
    accusamus reiciendis neque itaque nam, vel tenetur sit perspiciatis doloremque 
    voluptatibus ex rem non officia aut nulla corporis, dicta ducimus.`,
    author: 'Jake',
    _id: 'string',
    likes: 21,
    comments: [],
    views: 123,
    image: 'image.jpg',
};

describe('PostData', () => {
    it('render with post data', () => {
        const component = render(
            <PostData activePost={activePost} />
        );
        expect(screen.getByText(/one/i)).toBeInTheDocument();
        expect(screen.getByText(/two/i)).toBeInTheDocument();
        expect(screen.getByText(/автор: jake/i)).toBeInTheDocument();
        expect(screen.getByText(/anime/i)).toBeInTheDocument();
        expect(screen.getByText(/lorem ipsum dolor/i)).toBeInTheDocument();

        expect(component).toMatchSnapshot();
    });
});