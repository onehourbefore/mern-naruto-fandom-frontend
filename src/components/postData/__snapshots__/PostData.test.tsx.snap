// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostData render with post data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root__post"
      >
        <div
          class="root__header"
        >
          <div
            class="root__post_author"
          >
            <span
              class="root__post_author_tags"
            >
              one
            </span>
            <span
              class="root__post_author_tags"
            >
              two
            </span>
          </div>
          <div
            class="root__post_author"
          >
            Автор: 
            Jake
          </div>
        </div>
        <h2
          class="root__post_title"
        >
          ANIME
        </h2>
        <div
          class="root__post_content"
        >
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. In quisquam 
    accusamus reiciendis neque itaque nam, vel tenetur sit perspiciatis doloremque 
    voluptatibus ex rem non officia aut nulla corporis, dicta ducimus.
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root__post"
    >
      <div
        class="root__header"
      >
        <div
          class="root__post_author"
        >
          <span
            class="root__post_author_tags"
          >
            one
          </span>
          <span
            class="root__post_author_tags"
          >
            two
          </span>
        </div>
        <div
          class="root__post_author"
        >
          Автор: 
          Jake
        </div>
      </div>
      <h2
        class="root__post_title"
      >
        ANIME
      </h2>
      <div
        class="root__post_content"
      >
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. In quisquam 
    accusamus reiciendis neque itaque nam, vel tenetur sit perspiciatis doloremque 
    voluptatibus ex rem non officia aut nulla corporis, dicta ducimus.
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
